kolibri/utils/sanity_checks.py

```diff
@@ -24,26 +26,19 @@ def check_other_kolibri_running(port):
     except NotRunning:
         # In case that something other than Kolibri occupies the port,
         # check the port's availability.
-        check_port_availability('127.0.0.1', port)
+        check_port_availability(LISTEN_ADDRESS, port)
 
 
 def check_port_availability(host, port):
-    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    # This is to prevent the previous execution has left the socket
-    # in a TIME_WAIT start, and can't be immediately reused.
-    # From the bottom of https://docs.python.org/2/library/socket.html#example
-    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     try:
-        s.bind((host, port))
-        s.close()
-    except socket.error:
+        portend.free(host, port, timeout=2)
```

Comment:
Is the 2 second timeout an informed choice? Could you add it as a constant at the top, like `PORT_AVAILABILITY_CHECK_TIMEOUT = 2` ?

Otherwise, maybe just lave it out until we know if restarting Kolibri can leave a port occupied and for how long in that case.. we can always re-add it.